monai/losses/barlow_twins.py

```diff
@@ -0,0 +1,83 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from warnings import warn
+
+import torch
+from torch.nn.modules.loss import _Loss
+
+
+class BarlowTwinsLoss(_Loss):
+    """
+    Compute the Barlow Twins loss defined in:
```

Comment:
Thanks for the contribution! Could you please provide some explanation here in the docstring on what the loss is for, what it computes, what applications it would be for, example use case if there's something particular to demonstrate, etc.? It helps at a glance to know what this loss would be for without reading the paper.

Reply:
Yes, of course!
I've modified the doctring to provide a little more explanation of the usefulness of this loss function.

Reply:
Thanks for this. I think one of the key things here is that it's a loss on the cross-correlation matrix rather than on the images directly. I haven't used this before myself so I would benefit from an example of how this would be used in MONAI. If you're planning a tutorial notebook to show how to use this loss that would go there (I hope you're planning on that!), otherwise something added here to explain that technical detail would help.

I had made minor suggestions in the tests but otherwise it looks good to me.

Reply:
This is similar to the contrastive cost function for pre-training encoders.
Yes, perhaps a short tutorial would be more helpful. If I'm not being silly, the MONAI tutorials are on another repo though, should I make a PR on this other repo with a tutorial?

Reply:
Hello @ericspod. I have created a notebook tutorial for this loss self supervised as requested. If this PR is merged I will push a PR in the tutorial repo with the corresponding notebook.

Binder: https://mybinder.org/v2/gh/Lucas-rbnt/monai_tutorial/main
Directly on my repo: https://github.com/Lucas-rbnt/monai_tutorial/blob/main/ssl_barlow_twins.ipynb

Reply:
Looks good, thanks. When you do the PR to the tutorials repo we can look at the notebook in more depth, but I would suggest now that it should also have more description on what it's doing step-by-step. You should emphasise what's the main focus or concept you're demonstrating so it doesn't get lost in all the code for training and such. 

 I'm fine with this right now but @Nic-Ma @KumoLiu thoughts?

Reply:
Agreed, we can discuss more about the notebook after you create a PR.